README.md

```diff
@@ -1,10 +1,568 @@
+    participant ACS Chat SDK
+    participant Azure Communication Service
+    Web App-->>App Web Server: Send User Token for authentication and ask for an ACS token
+    App Web Server -->> App Web Server: Validate the user using User Token
+    App Web Server -->> ACS Token Service: Create an ACS token
+    ACS Token Service -->> App Web Server: Return an ACS token
+    App Web Server -->> App Web Server: Bind User Identity with ACS token
+    App Web Server -->> Web App: Return an ACS token
+    Web App -->> ACS Adapter: Use ACS token to create adapter
+    Web App -->> WebChat: Render Webchat and bind with adapter
+    ACS Adapter -->> ACS Chat SDK: Create ChatClient using ACS token
+    ACS Chat SDK -->> Azure Communication Service: Network request to service
+```
+
+Notes:
+
+1. All authentications depend on the original app authentication, and the application server is responsible for assigning a token after user validation (and binding the token with user info)
+2. Every time the token service is asked to create a new user token, a new communications services user is created and token binding happens again
+
+3. Thread creation could happen either in the backend service or in the adapter; it is a preference of the developer
+
+## Message sending and receiving
+
+```mermaid
+sequenceDiagram
+    participant WebChat
+    participant ACS Adapter
+    participant ACS Chat SDK
+    participant Azure Communication Service
+    WebChat -->> WebChat: User sends messages
+    WebChat -->> ACS Adapter: Pass message as an Activity to ACS Adapter
+    ACS Adapter -->> ACS Chat SDK: Call chatClient.sendMessage(message)
+    ACS Chat SDK -->> Azure Communication Service: Send post message network request
+    Azure Communication Service -->> ACS Chat SDK: Receive message from network
+    ACS Chat SDK -->> ACS Adapter: Trigger receive message event
+    ACS Adapter -->> WebChat: Dispatch IncomingActivity
+```
+
+## Details for waiting queue
+
+```mermaid
+sequenceDiagram
+    participant ACS Adapter
+    participant ChatWidget
+    participant Edge Server
+    participant Azure Communication Service
+    Edge Server -->> Azure Communication Service: Join Conversation (using ACS Chat SDK)
+    Edge Server -->> Azure Communication Service: Create token and add customer into thread
+    Edge Server -->> ChatWidget: token & threadId
+    ChatWidget -->> ACS Adapter: token & threadId
+    Edge Server -->> Azure Communication Service: Send queue information (using chatClient.sendMessage)
+    Azure Communication Service -->> ACS Adapter: Dispatch queue messgae
+    ACS Adapter -->> ChatWidget: Dispatch Activity with ChannelData
+    Edge Server -->> Azure Communication Service: Agent ready, create token and join agent to thread
+    Edge Server -->> Azure Communication Service: Agent ready, create token and join the agent to the thread
+    Azure Communication Service -->> ACS Adapter: trigger threadUpdate
+```
+
+## Details for idle status
+
+```mermaid
```

Comment:
Should we delete this? (I don't think it's a part of the functionslity right now)